trestle/core/commands/author/ssp.py

```diff
@@ -97,44 +83,83 @@ def _run(self, args: argparse.Namespace) -> int:
-            if args.sections:
-                sections_dict = sections_to_dict(args.sections)
-                if const.STATEMENT in sections_dict:
-                    raise TrestleError('Statement is not allowed as a section name.')
-                # add any existing sections from the controls but only have short names
-                control_section_short_names = catalog_interface.get_sections()
-                for short_name in control_section_short_names:
-                    if short_name not in sections_dict:
-                        sections_dict[short_name] = short_name
-                logger.debug(f'ssp sections dict: {sections_dict}')
-
-            context = ControlContext.generate(ContextPurpose.SSP, True, trestle_root, markdown_path)
-            context.yaml_header = yaml_header
-            context.sections_dict = sections_dict
-            context.prompt_responses = True
-            context.overwrite_header_values = args.overwrite_header_values
-            context.allowed_sections = args.allowed_sections
-
-            catalog_interface.write_catalog_as_markdown(context, catalog_interface.get_statement_part_id_map(False))
-
-            return CmdReturnCodes.SUCCESS.value
 
         except Exception as e:  # pragma: no cover
             return handle_generic_command_exception(e, logger, 'Error while writing markdown from catalog')
 
+    def _generate_ssp_markdown(
+        self,
+        trestle_root: pathlib.Path,
+        profile_path: pathlib.Path,
+        compdef_name_list: List[str],
+        md_path: pathlib.Path,
+        yaml_header: Dict[str, Any],
+        overwrite_header_values: bool,
+        force_overwrite: bool
+    ) -> int:
+        """
+        Generate the ssp markdown from the profile and compdefs.
+
+        Notes:
+        Get RPC from profile.
+        For each compdef:
+            For each comp:
+                Load top level rules
+                for each control_imp:
+                    Load rules params values
+                    For each imp_req (bound to 1 control):
+                        Load control level rules and status
+                        Load part level rules
+                        If rules apply then write out control and add to list written out
+                        If control exists, read it and insert content
+        """
+        if force_overwrite:
+            try:
+                logger.debug(f'Overwriting the content of {md_path}.')
+                clear_folder(pathlib.Path(md_path))
+            except TrestleError as e:  # pragma: no cover
+                raise TrestleError(f'Unable to overwrite contents of {md_path}: {e}')
+
+        context = ControlContext.generate(ContextPurpose.SSP, True, trestle_root, md_path)
+        context.cli_yaml_header = yaml_header
+        context.sections_dict = {}
+        context.prompt_responses = True
+        context.overwrite_header_values = overwrite_header_values
+        context.allowed_sections = []
```

Comment:
Let section_dict and allowed_sections be `None`, the users should be able to handle it